libpod/networking_linux.go

```diff
@@ -134,6 +134,12 @@ func (r *RootlessNetNS) Do(toRun func() error) error {
 		if err != nil {
 			return errors.Wrap(err, "could not get runtime directory")
 		}
+		// eval symlinks since they may not exists in the new path: https://github.com/containers/podman/issues/14606
+		xdgRuntimeDir, err = filepath.EvalSymlinks(xdgRuntimeDir)
```

Comment:
would it work to have the check once only in `SetXdgDirs()` instead of two different places?